File: careerpilot-frontend/types/job.ts
export interface Job {
  id?: number;

  job_id?: string;

  title: string;
  company: string;
  location: string | null;
  description: string;

  salary?: string | null;

  job_type?: string | null;
  employment_type?: string | null;

  experience_required?: number | null;

  apply_link?: string;
  logo?: string | null;

  created_at?: string;
}

export interface JobMatchResult {
  match_score?: number;
  matched_skills?: string[];
  missing_skills?: string[];
  strengths?: string[];
  suggestions?: string[];
  error?: string;
  raw_response?: string;
}

export interface JobFilters {
  search?: string;
  location?: string;
  job_type?: string;
}

File: backend/app/Schemas/job.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class JobBase(BaseModel):
    title: str
    company: str
    location: str | None = None
    description: str
    salary: str | None = None
    job_type: str | None = None
    experience_required: int | None = None


class JobCreate(JobBase):
    pass


class JobUpdate(BaseModel):
    title: str | None = None
    company: str | None = None
    location: str | None = None
    description: str | None = None
    salary: str | None = None
    job_type: str | None = None
    experience_required: int | None = None


class JobResponse(JobBase):
    id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)